src/components/Traceroute/Traceroute.tsx

```diff
@@ -1,52 +1,40 @@
-    "If you chose custom scan, you can type your customized traceroute option.\n" +
-    "For example: -I -p 12345\n\n" +
-    "Step 1: Enter a Hostname/IP address.\n" +
-    "       Eg: 192.168.0.1\n\n" +
-    "Step 2: (Optional) Enter any additional options.\n\n" +
-    "Step 3: Select a scan option.\n" +
-    "       Eg: Traceroute UDP scan\n\n" +
-    "Step 4: Click Scan to commence Traceroute operation.\n\n" +
-    "Step 5: View the Output block below to view the results of the tool's execution.";
-
-//Variables
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import { RenderComponent } from "../UserGuide/UserGuide";
+
+/**
+ * FormValuesType defines the structure for the form values used in the TracerouteTool component.
+ * @field hostname: The hostname or IP address for the traceroute operation.
+ * @field portNumber:The port number to be used, currently not utilized in the traceroute operations.
+ * @field tracerouteSwitch: The selected type of traceroute scan.
+ * @field tracerouteOptions: Custom traceroute options provided by the user.
+ */
 interface FormValuesType {
     hostname: string;
     portNumber: string;
     tracerouteSwitch: string;
     tracerouteOptions: string;
 }
 
-//Traceroute Options
-const tracerouteSwitch = [
-    "Traceroute ICMP scan",
-    "Traceroute TCP scan",
-    "Traceroute UDP scan",
-    "Traceroute custom scan",
-];
-
-//Tool name must be capital or jsx will cry out errors :P
+// Component State Variables.
 const TracerouteTool = () => {
-    var [output, setOutput] = useState("");
-    const [selectedScanOption, setSelectedTracerouteOption] = useState("");
-
+    var [output, setOutput] = useState(""); //State to store the output from the traceroute command
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [selectedScanOption, setSelectedTracerouteOption] = useState(""); // State to store the selected scan type.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the model is opened.
+    const [LoadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the model.
+    const [allowSave, setAllowSave] = useState(false); // State variable indicating whether the current state is valid and the results can be saved.
+    const [hasSaved, setHasSaved] = useState(false); // State variable that tracks whether the results have already been saved to avoid redundant operations.
+    // Form hook to handle form input.
     let form = useForm({
```

Comment:
Doc String

Reply:
Update to camelCase

Reply:
Not sure which one you are referring to here.